packages/appkit/exports/react.ts

```diff
@@ -60,3 +61,11 @@ export function useAppKitNetwork(): UseAppKitNetworkReturn {
 }
 
 export { useAppKitAccount } from '@reown/appkit-core/react'
+
+export async function useAppKitBalance() {
+  if (!modal) {
+    throw new Error('AppKit not initialized. Please call createAppKit first.')
+  }
+
+  return await fetchBalance(modal)
```

Comment:
Do we need await ?

```ts
return fetchBalance(modal)
```

Reply:
Imo we should return this function in an object. Later there might be other properties or functions we might want to expose from this hook and we wouldn't want it to be breaking change. 

Also to follow the same convention with other hooks

```suggestion
 function fetchBalance() {
  return fetchBalance(modal)
 }
 
  return {
   fetchBalance
  }
```

Reply:
Also if I'm not wrong they will send request for fetchBalance as soon as the they call this hook. They cannot control the fetch time with this way right?

```jsx

function MyComp () {
   const balance = useAppKitBalance() // this will immediately call the balance, which would be better to let responsiblity to the developers imo
...

}
```

Reply:
+1 to Enes' point. All of our hooks usually return handlers, not responses